client/src/components/Forms/EventForm.tsx

```diff
@@ -0,0 +1,92 @@
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import React, { useState } from 'react';
+import { useUser } from '../../hooks/useUser';
+import { showToast } from '../../utils/showToast';
+import getAuthErrorMessage from '../../utils/getAuthErrorMessage';
+import TextField from '@mui/material/TextField';
+import Button from '../../ui/Button/Button';
+import eventsApi from '../../api/eventsApi';
+import { NewEvent } from '../../types/events';
+
+const NewEventForm = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const { userId } = useUser();
+  const maxDate = new Date();
+  maxDate.setFullYear(maxDate.getFullYear() + 1);
+
+  const formik = useFormik({
+    initialValues: {
+      eventName: '',
+      description: '',
+      eventDate: '',
+      userID: '',
+    },
+    validationSchema: Yup.object().shape({
+      eventName: Yup.string().required('Name is required').max(30, 'Name is too long'),
+      description: Yup.string(),
+      eventDate: Yup.date()
+        .required('Deadline is required')
+        .min(new Date(), 'Event cannot be planned for the past.')
+        .max(maxDate, 'Event is too far in the future'),
+    }),
+    validateOnChange: true,
+    onSubmit: async (values) => {
+      try {
+        setIsLoading(true);
+        await eventsApi.addEvent({ ...values, userID: userId } as NewEvent);
```

Comment:
Type assertion shouldnt be used here the whole object is good typed and ts knows that. Problem here is with userId, ts think there could be undefined so you should check if its not undefined or if you are 100% sure about userId than its not undefined then you can use nonNon-null Assertion Operator.
[source](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-)

Type assertions are good for ts if he dont know the types like some html elements or maybe some responses from be the most common way is just check if its not undefined but like I said i'm using non null assertion when I'm 100% sure.